climada/entity/exposures/base.py

```diff
@@ -1121,20 +1122,31 @@ def plot_basemap(
         self.to_crs(crs_ori, inplace=True)
         return axis
 
-    def write_hdf5(self, file_name):
+    def write_hdf5(self, file_name, pickle_geometry_as_shapely=False):
         """Write data frame and metadata in hdf5 format
 
         Parameters
         ----------
         file_name : str
             (path and) file name to write to.
+        pickle_geometry_as_shapely : bool
+            flag, indicating whether the "geometry" of the Exposures` `data` will be stored as
+            pickled shapely objects instead of wkb bytes. This has been the case for earlier
+            CLIMADA version, up to 6.0, and is perhaps faster but less durable,
+            because pickled data may evantually get unreadable for future shapely versions.
+            Default: False
         """
         LOGGER.info("Writing %s", file_name)
         store = pd.HDFStore(file_name, mode="w")
         pandas_df = pd.DataFrame(self.gdf)
+        wkb_columns = []
         for col in pandas_df.columns:
             if str(pandas_df[col].dtype) == "geometry":
-                pandas_df[col] = np.asarray(self.gdf[col])
+                if pickle_geometry_as_shapely:
+                    pandas_df[col] = np.asarray(self.gdf[col])
+                else:
+                    pandas_df[col] = gpd.GeoSeries.to_wkb(pandas_df[col])
```

Comment:
`GeoSeries.to_wkb` calls `shapely.to_wkb` on `self`.  Are you sure this does what it's supposed to do?
```suggestion
                    pandas_df[col] = gpd.GeoSeries(pandas_df[col]).to_wkb()
```
It might be more elegant to use [`GeoDataFrame.to_wkb`](https://geopandas.org/en/stable/docs/reference/api/geopandas.GeoDataFrame.to_wkb.html) on `self`, which already returns a `pandas.DataFrame`.

Reply:
Thanks for the review!

>  Is there a particular reason why we need to keep the old behavior of write_hdf5 at all?

Two reasons:
- I'm afraid the new way is slower, so the old way would be preferable if files don't have to be kept forever.
- For testing readability of old style files.

Reply:
Because you are calling a method of an uninstantiated object. Maybe use `shapely.to_wkb(pandas_df[col])` directly?

Reply:
I'll keep it as it is (`pandas_df[col] = gpd.GeoSeries(pandas_df[col]).to_wkb()`) so geopandas can take care of it.